File: data/StarCoder2/readable_format_HPEC/fortran/file184.f90
subroutine bubble( ten, j1, j2, j3, m, ind )


implicit none


integer m, ind, j1( m, 0:1 ), j2( m, 0:1 ), j3( m, 0:1 )
double precision ten( m, 0:1 )
double precision temp
integer i, j_temp

if( ind .eq. 1 )then

do  i=1,m-1
if( ten(i,ind) .gt. ten(i+1,ind) )then

temp = ten( i+1, ind )
ten( i+1, ind ) = ten( i, ind )
ten( i, ind ) = temp

j_temp           = j1( i+1, ind )
j1( i+1, ind ) = j1( i,   ind )
j1( i,   ind ) = j_temp

j_temp           = j2( i+1, ind )
j2( i+1, ind ) = j2( i,   ind )
j2( i,   ind ) = j_temp

j_temp           = j3( i+1, ind )
j3( i+1, ind ) = j3( i,   ind )
j3( i,   ind ) = j_temp

else
return
endif
enddo

else

do  i=1,m-1
if( ten(i,ind) .lt. ten(i+1,ind) )then

temp = ten( i+1, ind )
ten( i+1, ind ) = ten( i, ind )
ten( i, ind ) = temp

j_temp           = j1( i+1, ind )
j1( i+1, ind ) = j1( i,   ind )
j1( i,   ind ) = j_temp

j_temp           = j2( i+1, ind )
j2( i+1, ind ) = j2( i,   ind )
j2( i,   ind ) = j_temp

j_temp           = j3( i+1, ind )
j3( i+1, ind ) = j3( i,   ind )
j3( i,   ind ) = j_temp

else
return
endif
enddo

endif

return
end

File: data/gpt-4o/readable_format_HPEC/cpp/file184.cpp
void bubble(double ten[][2], int j1[][2], int j2[][2], int j3[][2], int m, int ind){
 double temp;
 int i, j_temp;
 if(ind == 1){
  for(i=0; i<m-1; i++){
   if(ten[i][ind] > ten[i+1][ind]){
    temp = ten[i+1][ind];
    ten[i+1][ind] = ten[i][ind];
    ten[i][ind] = temp;

    j_temp = j1[i+1][ind];
    j1[i+1][ind] = j1[i][ind];
    j1[i][ind] = j_temp;

    j_temp = j2[i+1][ind];
    j2[i+1][ind] = j2[i][ind];
    j2[i][ind] = j_temp;

    j_temp = j3[i+1][ind];
    j3[i+1][ind] = j3[i][ind];
    j3[i][ind] = j_temp;
   }
   else return;
  }
 }
 else{
  for(i=0; i<m-1; i++){
   if(ten[i][ind] < ten[i+1][ind]){
    temp = ten[i+1][ind];
    ten[i+1][ind] = ten[i][ind];
    ten[i][ind] = temp;

    j_temp = j1[i+1][ind];
    j1[i+1][ind] = j1[i][ind];
    j1[i][ind] = j_temp;

    j_temp = j2[i+1][ind];
    j2[i+1][ind] = j2[i][ind];
    j2[i][ind] = j_temp;

    j_temp = j3[i+1][ind];
    j3[i+1][ind] = j3[i][ind];
    j3[i][ind] = j_temp;
   }
   else return;
  }
 }
}
